runtime/ops/process.rs

```diff
@@ -337,17 +331,139 @@ fn create_command(
+      OPEN_EXISTING, PIPE_ACCESS_DUPLEX,
+    };
+    use windows_sys::Win32::System::Pipes::ConnectNamedPipe;
+    use windows_sys::Win32::System::Pipes::{
+      CreateNamedPipeW, PeekNamedPipe, PIPE_READMODE_BYTE,
+      PIPE_READMODE_MESSAGE, PIPE_REJECT_REMOTE_CLIENTS, PIPE_TYPE_BYTE,
+      PIPE_TYPE_MESSAGE, PIPE_UNLIMITED_INSTANCES,
+    };
+
+    use std::io;
+    use std::os::windows::ffi::OsStrExt;
+    use std::path::Path;
+    use std::ptr;
+
+    if let Some(ipc) = args.ipc {
+      if ipc < 0 {
+        return Ok((command, None));
+      }
+
+      let name = format!("\\\\.\\pipe\\{}", uuid::Uuid::new_v4());
+      let mut path = Path::new(&name)
+        .as_os_str()
+        .encode_wide()
+        .collect::<Vec<_>>();
+      path.push(0);
+
+      // TODO(@littledivy): Handle pipe name collisions
+
+      let hd1 = CreateNamedPipeW(
+        path.as_ptr(),
+        PIPE_ACCESS_DUPLEX
+          | FILE_FLAG_FIRST_PIPE_INSTANCE
+          | FILE_FLAG_OVERLAPPED,
+        PIPE_TYPE_BYTE | PIPE_READMODE_BYTE,
+        1,
+        65536,
+        65536,
+        0,
+        std::ptr::null_mut(),
+      );
+
+      if hd1 == INVALID_HANDLE_VALUE {
+        return Err(std::io::Error::last_os_error().into());
+      }
+
+      /* Create child pipe handle. */
+      let mut s = SECURITY_ATTRIBUTES {
+        nLength: std::mem::size_of::<SECURITY_ATTRIBUTES>() as u32,
+        lpSecurityDescriptor: ptr::null_mut(),
+        bInheritHandle: 1,
+      };
+      let mut hd2 = CreateFileW(
+        path.as_ptr(),
+        GENERIC_READ | GENERIC_WRITE,
+        0,
+        &mut s,
+        OPEN_EXISTING,
+        FILE_FLAG_OVERLAPPED,
+        0,
+      );
+      if hd2 == INVALID_HANDLE_VALUE {
+        return Err(io::Error::last_os_error().into());
+      }
+
+      // Will not block because we have create the pair.
+      if ConnectNamedPipe(hd1, ptr::null_mut()) == 0 {
+        let err = std::io::Error::last_os_error();
+        if err.raw_os_error() != Some(ERROR_PIPE_CONNECTED as i32) {
+          CloseHandle(hd2);
+          return Err(err.into());
+        }
+      }
+
+      use windows_sys::Win32::Foundation::DuplicateHandle;
+      use windows_sys::Win32::Foundation::DUPLICATE_SAME_ACCESS;
+      use windows_sys::Win32::System::Threading::GetCurrentProcess;
+
+      // Duplicating the handle to allow the child process to use it.
+      if DuplicateHandle(
+        GetCurrentProcess(),
+        hd2,
+        GetCurrentProcess(),
+        &mut hd2,
+        0,
+        1,
+        DUPLICATE_SAME_ACCESS,
+      ) == 0
+      {
+        return Err(std::io::Error::last_os_error().into());
+      }
+
+      /* One end returned to parent process (this) */
+      let pipe_fd = Some(
+        state
+          .resource_table
+          .add(deno_node::IpcJsonStreamResource::new(hd1 as i64)?),
+      );
+
+      /* The other end passed to child process via DENO_CHANNEL_FD */
+      command.env("DENO_CHANNEL_FD", format!("{}", hd2 as i64));
```

Comment:
Now that I think of it, we should probably use `get_osfhandle` and `open_osfhandle`.  It seems it is also safe to discard the upper 32bits of the handle as per Microsoft https://learn.microsoft.com/en-us/windows/win32/winprog64/interprocess-communication?redirectedfrom=MSDN

Reply:
It seems we cannot use `get_osfhandle` because Deno statically links to msvcrt. I've added a comment in `ipc.rs`